src/_tkagg.cpp

```diff
@@ -92,8 +100,13 @@ static PyObject *mpl_tk_blit(PyObject *self, PyObject *args)
     block.offset[1] = o1;
     block.offset[2] = o2;
     block.offset[3] = o3;
-    TK_PHOTO_PUT_BLOCK_NO_COMPOSITE(
-        photo, &block, x1, height - y2, x2 - x1, y2 - y1);
+    put_retval = TK_PHOTO_PUT_BLOCK(
+        interp, photo, &block, x1, height - y2, x2 - x1, y2 - y1, comp_rule);
+    Py_END_ALLOW_THREADS
+    if (put_retval) {
```

Comment:
I guess this should check if it's `!= TCL_OK`, but we haven't had any other cases of `TCL_OK`?

Reply:
Right, and the only reason the function call is not within the if parens is that the error response needs the GIL. But since you bring it up, putting TCL_OK in the headers isn't going to be a burden.

Reply:
also only TCL_ERROR is possible according to the docs